generator/camera_info_collector.cpp

```diff
@@ -137,6 +136,7 @@ bool CamerasInfoCollector::ParseIntermediateInfo(
                    "lat(", lat, "), lon(", lon, ")"));
     }
 
+    std::vector<routing::SpeedCameraMwmPosition> ways;
```

Comment:
I am not sure what was intention here initially, so fixed it with my understanding of surrounding code, but please double check it.

Reply:
Just to better understand: `ways` are declared outside of loop and being moved inside this loop.

Reply:
Good catch. But the better fix would be to call `ways.clear()` to explicitly set the empty state, and to avoid unnecessary allocations in each loop iteration.

Reply:
Well, I'd prefer to declare ways vector as close to the using scope as possible.

Reply:
Then buffer_vector can be a better candidate. Avoiding unnecessary allocations is a very good pattern.

Reply:
This is generator, so no need to check anything here. And buffer_vector also should treat carefully (it sucks sometimes) :)

Reply:
_Faster_ generator is also a great thing )

Reply:
You won't get anything faster. Code readability is much better here.

Reply:
Added move for the honest comparison. Clear is 3x-5x times faster than reassign. I'm still strongly voting for the faster code. Clear is very _clear_ for readability.

https://godbolt.org/z/9hsWdeKGr

Reply:
I don't mind. But looks like benchmark was in debug :) Here is my results:
<img width="308" alt="Screenshot 2022-12-17 at 18 56 25" src="https://user-images.githubusercontent.com/175612/208255100-81abcc91-ec4a-4549-bb07-1b16c40fcf98.png">

The actual dispersion is bigger than actual timings ..

Reply:
You're right, in the second link the -O3 switch was reset. You need to add -std=c++17 btw for the correct code generation.
Then results are not as straight, and depend on the compiler (clang is better on linux than gcc):
```
Clear 0
Clear 0
Clear 41
Clear 83
Clear 0
Clear 0
Reassign 209
Reassign 167
Reassign 41
Reassign 41
Reassign 42
Reassign 42
```